### ISSUE TITLE
[C++11] member pointer declaration cannot use decltype specifier

### ISSUE THREAD

**Issue by e48d0f12-b0d3-485a-93ce-75b823b0ccac** [llvm/llvm-project#21364]
|  |  |
| --- | --- |
| Bugzilla Link | [20990](https://llvm.org/bz20990) |
| Version | trunk |
| OS | All |
| CC | @DougGregor,@zygoloid |

## Extended Description 
The sample code below should be compiled successfully but it causes compilation error by clang.
```cpp
struct S {
    int i;
};

int main()
{
    S v;
    int decltype(v)::*p = &S::i;
}
```

The `decltype(v)::*` is `ptr-operator` and `decltype(v)::` is `nested-name-specifier`, so it is well-formed.

cf. http://melpon.org/wandbox/permlink/o2cISNE0JqJ2YbDj

**Comment by zwuis** [comment]
Doesn't reproduce since Clang 6.0.

**Comment by Endilll** [comment]
Appears to be fixed in Clang 3.1: https://godbolt.org/z/KM9jbchzh
